src/main/java/frc/robot/PowerController.java

```diff
@@ -0,0 +1,123 @@
+  private final double totalSustainableCurrent = 200.0;
+
+  private ControllableSubsystem[] subsystems;
+
+  public PowerController(ControllableSubsystem... subsystems) {
+    this.subsystems = subsystems;
+  }
+
+  public void check() {
+    System.out.println("Checking for a brownout...");
+
+    // If the PDP is close to a brownout, trigger brownout scaling
+    if (PDP.getVoltage() < 7) {
+      System.out.println("close to brownout!");
+      setBrownoutScaling();
+    }
+
+    // Sum up total current drawn from all subsystems
+    double totalCurrent = PDP.getTotalCurrent();
+    System.out.println("total current drawn: " + totalCurrent);
+
+    // If current goes over sustainable current, scale subsystems down
+    if (totalCurrent > totalSustainableCurrent) {
+      scale(totalCurrent, new HashSet<>());
+    }
+  }
+
+  /**
+   * Gets the total current drawn from the PDP by the specified channels.
+   * @param channels The channels to sum.
+   * @return The total current drawn by the provided channels.
+   */
+  public static double getCurrentDrawnFromPDP(int... channels) {
+    double sum = 0;
+
+    for (int channel : channels) {
+      sum += PDP.getCurrent(channel);
+    }
+
+    // Is this necessary?
+    if (channels.length == 16 && (sum != PDP.getTotalCurrent())) {
+      System.out.println("something is wrong, total current does not match");
+    }
+
+    return sum;
+  }
+
+  /**
+   * Scales all subsystems by a set sensible amount to prevent brownout.
+   * Call this as a last resort when the voltage drops to near brownout or if subsystem scaling fails to 
+   * maintain the sustainable current.
+   */
+  private void setBrownoutScaling() {
+    for (ControllableSubsystem subsystem : subsystems) {
+      double currDrawn = subsystem.getTotalCurrentDrawn();
+      int tempscale = (int) (currDrawn * 0.8);
+      subsystem.setCurrentLimit(tempscale);
+    }
+  }
+
+  /**
+   * Recursively scales down the lowest priority subsystem until the total drawn current falls below the
+   * sustainable threshold.
+   * @param current The current total current.
+   * @param checked A set of already scaled subsystems.
+   */
+  private void scale(double current, HashSet<ControllableSubsystem> checked) {
+    ControllableSubsystem lowestPriority = null;
+    int lowPriority = 10; // This int should be higher than the number of subsystems
+
+    // If we've already scaled many subsystems, just scale more dramatically for everything
+    if (checked.size() > 3) {
+      setBrownoutScaling();
+      return;
+    }
+
+    for (ControllableSubsystem subsystem : subsystems) {
+      int priority = checkPriority(subsystem);
+      if (priority < lowPriority && !(checked.contains(subsystem))) {
+        lowestPriority = subsystem;
+        lowPriority = priority;
+      }
+    }
+
+    double currDrawn = lowestPriority.getTotalCurrentDrawn();
+    lowestPriority.setCurrentLimit((int) lowestPriority.minCurrent());
```

Comment:
Seems pretty harsh, perhaps doing some math and scaling things proportionally would be better?